backend/internal_assembler/internal_assembler.ml

```diff
@@ -0,0 +1,255 @@
+
+let make_identification : Owee.Owee_elf.identification =
+  { elf_class = 2;
+    elf_data = 1;
+    elf_version = 1;
+    elf_osabi = 0;
+    elf_abiversion = 0
+  }
+
+let make_header identification shnum e_shoff : Owee.Owee_elf.header =
+  { e_ident = identification;
+    e_type = 1;
+    e_machine = 0x3E;
+    e_version = 1;
+    e_entry = 0L;
+    e_phoff = 0L;
+    e_shoff;
+    e_flags = 0;
+    e_ehsize = 64;
+    e_phentsize = 0;
+    e_phnum = 0;
+    e_shentsize = 64;
+    e_shnum = shnum;
+    e_shstrndx = shnum - 1
+  }
+
+let create_section name ~sh_type ~size ~offset ?align ?entsize ?flags ?sh_link
+    ?sh_info shstrtab =
+  let sh_addralign = Option.value ~default:1L align in
+  let sh_entsize = Option.value ~default:0L entsize in
+  let sh_flags = Option.value ~default:0L flags in
+  let sh_link = Option.value ~default:0 sh_link in
+  let sh_info = Option.value ~default:0 sh_info in
+  let sh_name_str = X86_proc.SectionName.name name in
+  let section : Owee.Owee_elf.section =
+    { sh_name = String_table.current_length shstrtab;
+      sh_type;
+      sh_flags;
+      sh_addr = 0L;
+      sh_offset = offset;
+      sh_size = size;
+      sh_link;
+      sh_info;
+      sh_addralign;
+      sh_entsize;
+      sh_name_str
+    }
+  in
+  String_table.add_string shstrtab sh_name_str;
+  section
+
+let make_section sections name ~sh_type ~size ?align ?entsize ?flags ?sh_link
+    ?sh_info ?body shstrtab =
+  let section : Owee.Owee_elf.section =
+    create_section name ~sh_type ~size
+      ~offset:(Section_table.current_offset sections)
+      ?align ?entsize ?flags ?sh_link ?sh_info shstrtab
+  in
+  Section_table.add_section sections name ?body section
+
+let make_text sections name raw_section align sh_string_table =
+  make_section sections name ~sh_type:1
+    ~size:(Int64.of_int (X86_binary_emitter.size raw_section))
+    ~flags:0x6L sh_string_table ~align
+    ~body:(X86_binary_emitter.contents_mut raw_section)
+
+let make_data sections name raw_section align sh_string_table =
+  make_section sections name ~sh_type:1
+    ~size:(Int64.of_int (X86_binary_emitter.size raw_section))
+    ~flags:0x3L sh_string_table ~align
+    ~body:(X86_binary_emitter.contents_mut raw_section)
+
+let make_shstrtab sections sh_string_table =
+  let name = ".shstrtab" in
+  make_section sections
+    (SectionName.from_name name)
+    ~sh_type:3
+    ~size:
+      (Int64.of_int
+         (String_table.current_length sh_string_table + 1 + String.length name))
+    sh_string_table
+
+let make_custom_section sections name raw_section sh_string_table =
+  let flags = X86_proc.SectionName.flags name in
+  let align = X86_proc.SectionName.alignment name in
+  make_section sections name
+    ~size:(Int64.of_int (X86_binary_emitter.size raw_section))
+    ~align ~flags
+    ~body:(X86_binary_emitter.contents_mut raw_section)
+    sh_string_table
+
+let make_relocation_section sections sym_tbl_idx relocation_table
```

Comment:
Likewise `sym_tbl_idx` should be labelled (please do a quick pass through the code looking for any other cases like this), since it's not obvious from the type of the parameter what it is doing, and the type of the parameter is such that we might well end up with other parameters of that type in the future.